heartbeat/hbtest/skima/results.go

```diff
@@ -0,0 +1,98 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package skima
+
+import "fmt"
+
+// Results the results of executing a schema.
+// They are a flattened map (using dotted paths) of all the values []ValueResult representing the results
+// of the IsDefs.
+type Results map[string][]ValueResult
+
+func (r Results) recordResult(path string, result ValueResult) {
+	if r[path] == nil {
+		r[path] = []ValueResult{result}
+	} else {
+		r[path] = append(r[path], result)
+	}
+}
+
+// EachResult executes the given callback once per Value result.
+// The provided callback can return true to keep iterating, or false
+// to stop.
+func (r Results) EachResult(f func(string, ValueResult) bool) {
+	kontinue := true
+	for path, pathResults := range r {
+		for _, result := range pathResults {
+			kontinue = f(path, result)
```

Comment:
```
if !f(path, result) {
    return
}
```
instead of `kontinue`?

Reply:
Lol, I forgot `return` was a thing. Good suggestion :-D